pkg/smb/nodeserver.go

```diff
@@ -200,6 +200,12 @@ func (d *Driver) NodeStageVolume(_ context.Context, req *csi.NodeStageVolumeRequ
 		mountOptions = mountFlags
 		if !gidPresent && volumeMountGroup != "" {
 			mountOptions = append(mountOptions, fmt.Sprintf("gid=%s", volumeMountGroup))
+			if !raiseGroupRWXInMountFlags(mountOptions, "file_mode") {
+				mountOptions = append(mountOptions, "file_mode=0774")
+			}
+			if !raiseGroupRWXInMountFlags(mountOptions, "dir_mode") {
+				mountOptions = append(mountOptions, "dir_mode=0775")
```

Comment:
if `dir_mode=0775` and `file_mode=0774` is required, why not set those mount options in storage class or pv?

Reply:
For dynamic provisioning case, the user doesn't create pv. The storage class can be used both by privileged users (who doesn't need fsGroup) and ordinary ones who expect that everything is fine if security context is set correctly for a pod. I think setting fsGroup for a pod must give access to the volume for any storage class. Is this wrong assumption?

Reply:
that's reasonable, and I think if `file_mode` is not specified in pv, then we should append `file_mode=0774`, same for dir_mode, if it's not specified in pv, then append `dir_mode=0775`, since if `file_mode`, `dir_mode` is specified in pv, we should respect the setting, that would make the logic more straightforward.

btw, in nfs volume mount, we use `SetOwnerShip` to recursively set the ownership of the volume per fsGroup, while this solution won't work for smb mount since the ownership cannot be changed after smb volume is mounted.

Reply:
> if file_mode, dir_mode is specified in pv, we should respect the setting

That conflicts with fsGroup implementation in Kubernetes and all other CSI drivers. If `fsGroup` in a Pod depends  on StorageClass parameters, it will need a very visible documentation that fsGroup support needs `file_mode: 0770` and `dir_mode: 0770` in a StorageClass or even make it a default.

I would prefer if the driver did the right thing out of the box, as this PR proposes.

Reply:
Hi @andyzhangx , we need somehow to unblock this PR. You wrote above:

>  if file_mode, dir_mode is specified in pv, we should respect the setting

This raises the question what must have priority: fsGroup from security context or file/dir_mode from pv. We believe that fsGroup must have priority because if the user sets it in his/her Pod's definition, he/she would expect access enabled for the volume, no matter how file/dir_mode set in other places. Jan raised a concern above:

> That conflicts with fsGroup implementation in Kubernetes and all other CSI drivers.

Can you tell us your opinion on that, thank you!

Reply:
could you add ut and fix the spelling issue in this PR? thx

Reply:
Surely, thanks!